jupyterlab-server-proxy/src/index.ts

```diff
@@ -1,57 +1,115 @@
-import { JupyterFrontEnd, JupyterFrontEndPlugin } from '@jupyterlab/application'; 
+import { JupyterFrontEnd, JupyterFrontEndPlugin, ILayoutRestorer } from '@jupyterlab/application';
 import { ILauncher } from '@jupyterlab/launcher';
 import { PageConfig } from '@jupyterlab/coreutils';
+import { IFrame, MainAreaWidget, WidgetTracker } from '@jupyterlab/apputils';
 
 import '../style/index.css';
 
+function newServerProxyWidget(id: string, url: string, text: string): MainAreaWidget<IFrame> {
+  const content = new IFrame({
+    sandbox: ['allow-same-origin', 'allow-scripts', 'allow-popups', 'allow-forms'],
+  });
+  content.title.label = text;
+  content.title.closable = true;
+  content.url = url;
+  content.addClass('jp-ServerProxy');
+  content.id = id;
+  const widget = new MainAreaWidget({ content });
+  widget.addClass('jp-ServerProxy');
+  return widget;
+}
 
-function addLauncherEntries(serverData: any, launcher: ILauncher, app: JupyterFrontEnd) {
-    for (let server_process of serverData.server_processes) {
-      let commandId = 'server-proxy:' + server_process.name;
+async function activate(app: JupyterFrontEnd, launcher: ILauncher, restorer: ILayoutRestorer) : Promise<void> {
+  const response = await fetch(PageConfig.getBaseUrl() + 'server-proxy/servers-info');
+  if (!response.ok) {
+    console.log('Fetching metadata about registered failed. Make sure jupyter-server-proxy is installed');
```

Comment:
Did you intend the message to say "registered servers failed" ? Also with a period at the end of "installed".

Reply:
... actually that message predates this PR, sorry!